lib/rules/func-style.js

```diff
@@ -65,6 +65,33 @@ module.exports = {
 		const { namedExports: exportFunctionStyle } = overrides;
 		const stack = [];
 
+		/**
+		 * Checks if a function declaration is part of an overloaded function
+		 * @param {ASTNode} node The function declaration node to check
+		 * @returns {boolean} True if the function is overloaded
+		 */
+		function isOverloadedFunction(node) {
+			const functionName = node.id.name;
+
+			if (node.parent.type === "ExportNamedDeclaration") {
+				return node.parent.parent.body.some(
+					member =>
+						member.type === "ExportNamedDeclaration" &&
+						member.declaration?.type === "TSDeclareFunction" &&
+						member.declaration.id.name === functionName,
+				);
+			}
+
+			return (
+				Array.isArray(node.parent.body) &&
```

Comment:
There's still no test that would hit this. I think those tests should have the "expression"` option.